### ISSUE TITLE
Adding a new attribute to an Embedded object breaks when it's not alphabetically last

### ISSUE THREAD

**Issue by djordjevics1998** [isar/isar#1155]
### Steps to Reproduce

1. Write a collection containing an embedded object.
2. Write at least one instance of the collection with an embedded object to the database
3. Add an attribute  to the embedded class that won't be alphabetically last.
4. Update the schema.
5. Run the app. It will fail during startup with an error. On my prod app it failed with an error: index out of range. On this code sample it failed with a message:
Unhandled Exception: RangeError (byteOffset): Invalid value: Not in inclusive range 0..18: 21
#0      _ByteDataView.getInt64 (dart:typed_data-patch/typed_data_patch.dart:5000:7)
#1      IsarReaderImpl._readLongOrNull (package:isar/src/native/isar_reader_impl.dart:149:29)
#2      IsarReaderImpl.readLongOrNull (package:isar/src/native/isar_reader_impl.dart:163:12)
#3      _embeddedModelDeserialize (package:isar_adding_attribute_issue/model_example.g.dart:375:15)
#4      IsarReaderImpl.readObjectOrNull (package:isar/src/native/isar_reader_impl.dart:268:23)
#5      _exampleModelDeserialize (package:isar_adding_attribute_issue/model_example.g.dart:74:27)
#6      IsarCollectionImpl.deserializeObject (package:isar/src/native/isar_collection_impl.dart:42:38)
#7      IsarCollectionImpl.deserializeObjects (package:isar/src/native/isar_collection_impl.dart:66:22)
#8      QueryImpl.findInternal.<anonymous closure> (package:isar/src/native/query_impl.dart:51:27)
<asynchronous suspension>
#9      IsarCommon._beginTxn (package:isar/src/common/isar_common.dart:54:18)
<asynchronous suspension>
#10     _MyHomePageState.initState.<anonymous closure> (package:isar_adding_attribute_issue/main.dart:43:30)
<asynchronous suspension>

### Code sample

// initializing sample data before adding attributes to the EmbeddedModel
final existingModels = await isar!.exampleModels.where().findAll();
      if(existingModels.isEmpty) {
        await isar!.writeTxn(() async { await isar!.exampleModels.putAll([const ExampleModel(embeddedModel: EmbeddedModel(a: 0, b: 1, c: 2)), ExampleModel(embeddedModel: EmbeddedModel(a: 3, b: 4, c: 5))]); });
      }

// schema

@collection
class ExampleModel {
  final Id id;
  final EmbeddedModel embeddedModel;

  const ExampleModel({this.id = Isar.autoIncrement, required this.embeddedModel});
}

@embedded
class EmbeddedModel {
  final int a, b, c;
  //final String? aString; // uncommenting this attribute won't work and i suspect it's because of the alphabetical order
  //final String? dString; // uncommenting this attribute will work

  const EmbeddedModel({this.a = 0/*, this.aString*//*, this.dString*/, this.b = 0, this.c = 0});
}

### Details

 - Platform: Linux
 - Flutter version: 3.7.7
 - Dart version: 2.19.4
 - Isar version: 3.0.5

---

 - I've published a complete project featuring this sample code on GitHub https://github.com/djordjevics1998/isar_adding_attribute_issue

**Comment by apastuhov** [comment]
Same issue. Also, seems to be related to https://github.com/isar/isar/issues/1093

Want to note that that bug totally corrups collection in DB. So even if you try to remove those fields - app will still fail.

And in debug mode DB inspector is not working. So there is no way to fix issue manually.

```
thread 'isarworker' panicked at 'range end index 49 out of range for slice of length 47', library/core/src/slice/index.rs:73:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Platform: MacOS
Flutter version: 3.3.10
Dart version: 2.18.6
Isar version: 3.0.5

**Comment by djordjevics1998** [comment]
Yes, it's the same bug. It's not only related to lists of Embedded objects, as I've demonstrated in my sample code. It also first happened to me with a list of embedded objects in the prod app.

The only workaround is to either name the attribute itself to be alphabetically last or to use @Name annotation which was proposed by a guy in the Telegram group (nicknamed with arab letters which i can't reciprocate here) to make it alphabetically last, which I've confirmed to work. So, the issue is with the name presentation of new embedded object attributes.

I don't have enough experience in the subject of recovering data from a corrupted collection to add anything. Test any changes on test collection before rolling it out to the public.

**For the time being, the only workaround is to make sure any new attributes are alphabetically last.**

**Comment by rahul-yr** [comment]
This is a major issue for production applications. This certainly limit the use of this db for business or enterprise applications. Any plans for fix ?

**Comment by omnitrogen** [comment]
We're releasing an app to production today with an Isar db embedded and we are quite worried about future updates in collection / embedded models since it seems to break the app.

Any followup on that issue?

**Comment by mlg-hub** [comment]
@omnitrogen i faced the same issue with our app in production. and it is a major issue ... the workaround is to create another embedded , but collection can be updated

**Comment by simc** [comment]
This is fixed in v3.1.0 which will be available soon on pub.dev.

**Comment by ismail-go** [comment]
Unfortunately I am still getting same issue v3.1.0.
